src/com/acivev/VibratorManager.kt

```diff
@@ -0,0 +1,176 @@
+package com.acivev
+
+import android.content.Context
+import android.os.Build
+import android.os.VibrationEffect
+import android.os.Vibrator
+import com.acivev.utils.VibrationSupport
+import com.acivev.utils.VibratorCheckUtils
+import ru.nsu.ccfit.zuev.osu.Config
+
+/**
+ * Manage vibration effects for Android 7+.
+ * It supports different vibration patterns and intensities.
+ * - Vibration intensity can be set dynamically.
+ */
+object VibratorManager {
+
+    private var vibrator: Vibrator? = null
+    private var hasVibrationSupport = false
+
+    /**
+     * Whether vibration for circles is enabled.
+     */
+    var isCircleVibrationEnabled = false
+        set(value) {
+            field = value
+            Config.setBoolean("vibrationCircle", value)
+        }
+
+    /**
+     * Whether vibration for sliders is enabled.
+     */
+    var isSliderVibrationEnabled = false
+        set(value) {
+            field = value
+            Config.setBoolean("vibrationSlider", value)
+        }
+
+    /**
+     * Whether vibration for spinners is enabled.
+     */
+    var isSpinnerVibrationEnabled = false
+        set(value) {
+            field = value
+            Config.setBoolean("vibrationSpinner", value)
+        }
+
+    /**
+     * The intensity of the vibration.
+     */
+    var intensity = 127
```

Comment:
It would be nice if this can default to the device's intensity (-1) instead of being configured by us, but that's for another day.